### ISSUE TITLE
change isDirty by removing from hasMany

### ISSUE THREAD

**Issue by miguelcobain** [danielspaniel/ember-data-change-tracker#23]
Thanks a lot for this project!
I know that removing/adding from hasMany relationships seems to reflect on the `.changed()` result, but does the same happen with the `isDirty` property?

```js
user.get('projects').removeObject(firstProject); // remove project1
user.changed() //=> {projects: true }
user.get('isDirty') //=> this returns false, but it should return true
```

I wonder if I'm doing some configuration wrong or if this isn't supposed to work.

**Comment by danielspaniel** [comment]
That was a really great bug you found. 
 How the hasMany ever worked in the past ( is a real puzzler ) because what I found was it should not have worked at all. Bizzzzzarrreeeee. 
 But if you get v0.6.1 ( newest release ) you will be back in business. 
 Let me know if you find any others.

**Comment by miguelcobain** [comment]
Updating indeed fixed it! Thanks!

**Comment by DavidPhilip** [comment]
@danielspaniel The fix directly accesses the hasMany relation via the content property:
`relationsObserver.push(descriptor.key + '.content.@each.id');`

This works fine with hasMany relations that are flagged as async (Promise wraps the loaded contents via this property) but doesn't if they are not.

I'll add a PR with a failing test. I suggest adding another definition for the `relationsObserver` skipping the content property. I can also do a PR with the fix if you like.

**Comment by danielspaniel** [comment]
Sure thing @DavidPhilip , I'd be happy to check out your fix, and you could do the  fix it in the same PR => #36  if you want. Makes it simpler I reckon.

**Comment by DavidPhilip** [comment]
Done
